### ISSUE TITLE
Universal Cookie Service

### ISSUE THREAD

**Issue by jeffwhelpley** [angular/angular#7302]
We need a universal service (i.e. both on client and server) that can be used by any Angular 2 app to manage cookies. This issue is for coming up with a design for this service and implementing it in core. Once that is in place, the server side implementation of CookieService will be implemented on the Angular Universal repo:

https://github.com/angular/universal/issues/273

**Comment by Binvention** [comment]
how would cookies be implemented on a server?

**Comment by manekinekko** [comment]
In many ways. Here is one https://github.com/pillarjs/cookies

**Comment by Binvention** [comment]
Okay? But that's not actually using cookies on the server it's just a way to send cookies to the browser.

**Comment by manekinekko** [comment]
You are right. I was just referring to this as an example of how we would implement the cookies API.

**Comment by jeffwhelpley** [comment]
Not sure if I am misinterpreting the question @Binvention but web server requests do have cookies that are accessible on the server. The point with this service (and the other universal ones I posted) are that these are critical things that pretty much everyone who is going to build a universal Angular 2 app is going to need. In this case, it would be something like this that would exist within a service of your universal app:

```
var jwt = myCookieService.get('jwt');
```

Now, on the client side this would behave just like the Angular 1.x $cookie, but on the server side it would be something that would get initialized by the target web server framework (ex. Hapi, Express, Koa, etc.). Something like this:

```
server.ext('onPreAuth', function (req, reply) {
  cookieService.init(req.cookie);
});
```

In any case, the goal of this issue is not to worry about the server side implementation. We will handle that on the Angular Universal repo. Rather it is to discuss having the interface for the cookie service either in core or platform so that everyone is using the same thing.

**Comment by jeffwhelpley** [comment]
Closing in favor of #7438.
